src/CMakeLists.txt

```diff
@@ -1,107 +1,81 @@
 # Must be included before CMAKE_INSTALL_INCLUDEDIR is used.
 include(GNUInstallDirs)
-set(${PROJECT_NAME}_installables "")
 
 if(SECP256K1_ASM STREQUAL "arm")
-  add_library(common OBJECT
+  target_sources(secp_common_objs PRIVATE
     asm/field_10x26_arm.s
   )
-  set(common_obj "$<TARGET_OBJECTS:common>")
-else()
-  set(common_obj "")
 endif()
 
-add_library(precomputed OBJECT
+add_library(secp_precomputed_objs OBJECT EXCLUDE_FROM_ALL
   precomputed_ecmult.c
   precomputed_ecmult_gen.c
 )
-set(internal_obj "$<TARGET_OBJECTS:precomputed>" "${common_obj}")
 
-add_library(secp256k1 SHARED EXCLUDE_FROM_ALL
-  secp256k1.c
-  ${internal_obj}
-)
-target_include_directories(secp256k1 INTERFACE
+add_library(secp_common_objs OBJECT EXCLUDE_FROM_ALL secp256k1.c)
```

Comment:
Why is `secp256k1.c` the source of the `secp_common_objs` library, not the `secp256k1` one?

Reply:
Yeah, I'm going to do away with common entirely as it's not needed now that I understand that we're linking benches/tests directly to shared libs. Will push up a cleaner version next week.